### PR TITLE
Add Nitro Enclave Status check to Heartbeat

### PR DESCRIPTION
**Pull request by arnav-deep**

### CODE REVIEW

File: src/aws.rs [hunk 1]
```diff
@@ -375,6 +439,56 @@ impl Aws {
...
+        match sess {
+            Ok(r) => {
+                let res = self
+                    .run_enclave(&r, image_url, req_vcpu, req_mem, bandwidth)
+                    .await;
+                match res {
+                    Ok(_) => {}
+                    Err(e) => {
+                        self.spin_down_instance(&instance_id, &job, region.clone())
+                            .await?;
```
Comment (nisarg-soni) [inline src/aws.rs:475]: `&instance_id` to instance_id

File: src/market.rs [hunk 2]
```diff
@@ -455,7 +509,43 @@ impl JobState {
...
+                                    )
+                                    .await;
+                                match res {
+                                    Ok(_) => {
+                                        println!(
+                                            "job {job}: enclave successfully ran on the instance"
+                                        );
+                                    }
+                                    Err(err) => {
+                                        println!("job {job}: failed to run enclave, {err}. Spinning instance down.");
```
Comment (nisarg-soni) [inline src/market.rs:539]: `schedule_termination` here? In case of repeat failure  to run enclave

File: src/market.rs [hunk 3]
```diff
@@ -455,7 +509,43 @@ impl JobState {
...
+                                    }
+                                }
+                            }
+                        }
+                        Err(err) => {
+                            println!("job {job}: failed to retrieve enclave state, {err}");
```
Comment (nisarg-soni) [inline src/market.rs:545]: same as above `schedule_termination` here?

File: src/aws.rs (hunk 1 above)
Comment (arnav-deep) [reply src/aws.rs:475]: Done.

File: src/market.rs (hunk 2 above)
Comment (arnav-deep) [reply src/market.rs:539]: The fn `run_enclave_on_instance` has `spin_down` in it already for every error

File: src/market.rs (hunk 3 above)
Comment (arnav-deep) [reply src/market.rs:545]: Same as above

**Review by nisarg-soni — APPROVED** [review]

**Pull request merged**